templates/lxc-oci.in

```diff
@@ -129,23 +129,40 @@ getenv() {
   return
 }
 
-# FIXME 1: only support numerical values in the configuration file.
-# FIXME 2: from the OCI image spec: "If group/gid is not specified,
-# the default group and supplementary groups of the given user/uid in
-# /etc/passwd from the container are applied."
-getuidgid() {
-  if [ "$#" -eq 0 ]; then
-    echo "0 0"
-    return
+# check var is decimal.
+isdecimal() {
+  var="$1"
+  if [ "${var}" -eq "${var}" ] 2> /dev/null; then
+    return 0
+  else 
+    return 1
   fi
+}
 
+# get uid, gid from oci image.
+getuidgid() {
   configpath="$1"
+  rootpath="$2"
+  passwdpath="${rootpath}/etc/passwd"
+  grouppath="${rootpath}/etc/group"
 
-  uidgid=$(jq -c -r '.config.User // "0:0"' < "${configpath}")
+  usergroup=$(jq -c -r '.config.User' < "${configpath}")
   # shellcheck disable=SC2039
-  uidgid=(${uidgid//:/ })
+  usergroup=(${usergroup//:/ })
+
+  user=${usergroup[0]:-0}
+  if ! isdecimal "${user}" && [ -f ${passwdpath} ]; then
+    user=$(grep "^${user}:" "${passwdpath}" | awk -F: '{print $3}')
+  fi
+
+  group=${usergroup[1]:-}
+  if [ -z "${group}" ] && [ -f "${passwdpath}" ]; then
+    group=$(grep "^[^:]*:[^:]*:${user}:" "${passwdpath}" | awk -F: '{print $4}')
+  elif ! isdecimal "${group}" && [ -f "${grouppath}" ]; then
+    group=$(grep "^${group}:" "${grouppath}" | awk -F: '{print $3}')
+  fi
```

Comment:
So the problem with this approach is that it assumes a system where /etc/passwd and /etc/group is used, this isn't guaranteed and depends on NSS configuration.

Any reason why you're not using `getent passwd` and `getent group` instead?